src/wasm.h

```diff
@@ -149,7 +149,8 @@ struct Literal {
     double f64;
   };
 
-  Literal() : type(WasmType::none), f64(0) {}
+  Literal() : Literal(WasmType::none) {}
+  explicit Literal(WasmType type) : type(type) { memset(&f64, 0, sizeof(f64)); }
```

Comment:
hmm, this can't work, can it? there is an overloading on WasmType and on int32_t, wouldn't the latter always win? why do we even need the `explicit` one? maybe I'm missing something?